### PR TITLE
update 2.0 public api in tensor

### PR DESCRIPTION
**Pull request by zhiboniu**
### PR types

Others
### PR changes

APIs
### Describe

update 2.0 public api in tensor

背景：
对于用户调用api的层级，由于之前没有明确的设计，api开放层级比较随意。影响用户使用中的逻辑和正规感受。
目的：
1）重新整理api开放层级，提升用户使用感受
2）API保留自有特色，同时更好地符合行业规范，降低用户学习成本
3）提高框架api逻辑性，便于后续维护和优化
主要实现方法：
1）__all__内容清晰化，内容逐个列出，一行一个。不再使用 __all__ 的向上传递机制，仅在公开API层级添加 __all__，其余的 __all__ 列表统一置空
2）不再使用 #DEFINE_ALIAS
3）导入内容清晰化，原则上不再使用 import *， 按需引入需要的API， 每行一个
4）原则上不再使用 from . import module， from .module import submodule时会引入module，避免重复引用
5）添加新的Public API时，需向上寻找到最近的节点添加 import 并加入 __all__ 列表
6）Public API列表由多个公开API层级的 __all__ 列表共同组成，文档抽取以此为准，这些指定层级是：
- paddle
- paddle.amp
- paddle.nn
- paddle.nn.functional
- paddle.nn.initializer
- paddle.nn.utils
- paddle.static
- paddle.static.nn
- paddle.io
- paddle.jit
- paddle.metric
- paddle.distribution
- paddle.optimizer
- paddle.optimizer.lr
- paddle.regularizer
- paddle.text
- paddle.vision
- paddle.callbacks
- paddle.distributed
- paddle.distributed.fleet
- paddle.distributed.fleet.base.distributed_strategy
- paddle.distributed.fleet.data_generator
- paddle.distributed.fleet.utils
- paddle.distributed.parallel
- paddle.distributed.utils
- paddle.sysconfig
- paddle.utils
- paddle.utils.download
- paddle.utils.profiler
修改示例展示：
原来写法：
image
新的写法：
image

对开发者使用API的影响：
在指定层级推荐方式：（不受影响）
  from paddle.nn import CTCLoss
  from paddle.nn import Conv2D
非指定层级支持但不推荐：（不受影响）
  from paddle.nn.layer.loss import CTCLoss
  from paddle.nn.layer.conv import Conv2D
对于import *使用方式不推荐。（非公开层级受影响）
  不再支持：（非公开层级）
  from paddle.nn.layer import *
  from paddle.nn.layer.conv import *
  from paddle.nn.initializer.norm import *
  from paddle.tensor.math import *
  from paddle.optimizer.adam import *
  可以支持：（公开层级）
  from paddle import *
  from paddle.nn import *
  from paddle.utils import *
  from paddle.optimizer import *

本pr相关修改对api的调用无明显影响

### CODE REVIEW

File: python/paddle/tensor/__init__.py [hunk 1]
```diff
@@ -11,205 +11,316 @@
...
-# from .tensor import LoDTensor        #DEFINE_ALIAS
-# from .tensor import LoDTensorArray        #DEFINE_ALIAS
-from .to_string import set_printoptions  #DEFINE_ALIAS
-
-from .array import array_length  #DEFINE_ALIAS
-from .array import array_read  #DEFINE_ALIAS
-from .array import array_write  #DEFINE_ALIAS
-from .array import create_array  #DEFINE_ALIAS
+#this list used in math_op_patch.py for _binary_creator_
+bina_lst = [ #noqa
```
Comment (willthefrog) [inline python/paddle/tensor/__init__.py]: rename to `binary_ops`?

File: python/paddle/tensor/math.py [hunk 2]
```diff
@@ -472,8 +399,8 @@ def remainder(x, y, name=None):
     return _elementwise_op(LayerHelper(op_type, **locals()))
 
 
-mod = remainder  #DEFINE_ALIAS
-floor_mod = remainder  #DEFINE_ALIAS
+mod = remainder  # noqa: F401
```
Comment (willthefrog) [inline python/paddle/tensor/math.py]: are these F401 errors? maybe F841?

**Review by willthefrog — CHANGES_REQUESTED** [review]

File: python/paddle/tensor/math.py (hunk 2 above)
Comment (zhiboniu) [reply python/paddle/tensor/math.py]: done

File: python/paddle/tensor/__init__.py (hunk 1 above)
Comment (zhiboniu) [reply python/paddle/tensor/__init__.py]: done

**Review by willthefrog — DISMISSED** [review]

File: python/paddle/tensor/math.py [hunk 3]
```diff
@@ -472,8 +399,8 @@ def remainder(x, y, name=None):
     return _elementwise_op(LayerHelper(op_type, **locals()))
 
 
-mod = remainder  #DEFINE_ALIAS
-floor_mod = remainder  #DEFINE_ALIAS
+mod = remainder  # noqa: F841
```
Comment (iclementine) [inline python/paddle/tensor/math.py:402]: 这么做貌似有问题啊。mod 和 remainder 的语义存在一些差别。

File: python/paddle/tensor/math.py [hunk 4]
```diff
@@ -472,8 +399,8 @@ def remainder(x, y, name=None):
...
+floor_mod = remainder  # noqa: F841
```
Comment (iclementine) [inline python/paddle/tensor/math.py:403]: 这个也是类似。floor_mod 和 remainder 的
目前这样的用例多吗？
有可能会有后续兼容性问题。

File: python/paddle/fluid/layers/math_op_patch.py [hunk 5]
```diff
@@ -370,13 +370,7 @@ def __impl__(self, other_var):
...
     else:
         import paddle.tensor
-        variabel_methods = paddle.tensor.linalg.__all__ + \
-                           paddle.tensor.math.__all__ + \
-                           paddle.tensor.logic.__all__ + \
-                           paddle.tensor.manipulation.__all__ + \
-                           paddle.tensor.search.__all__ + \
-                           paddle.tensor.stat.__all__ + \
-                           paddle.tensor.attribute.__all__
+        variabel_methods = paddle.tensor.binary_ops
```
Comment (iclementine) [inline python/paddle/fluid/layers/math_op_patch.py]: Only some of the methods of `paddle.Tensor` is created by `_binary_creator_` (those listed above as `variable_methods`), other methods are just the independent functions with the same name. 

These may not be binary methods.

I suggest changing the name.

File: python/paddle/tensor/__init__.py [hunk 6]
```diff
@@ -11,205 +11,315 @@
...
-# from .tensor import Tensor        #DEFINE_ALIAS
-# from .tensor import LoDTensor        #DEFINE_ALIAS
-# from .tensor import LoDTensorArray        #DEFINE_ALIAS
-from .to_string import set_printoptions  #DEFINE_ALIAS
-
-from .array import array_length  #DEFINE_ALIAS
-from .array import array_read  #DEFINE_ALIAS
-from .array import array_write  #DEFINE_ALIAS
-from .array import create_array  #DEFINE_ALIAS
+#this list used in math_op_patch.py for _binary_creator_
```
Comment (iclementine) [inline python/paddle/tensor/__init__.py:185]: They are not used by `_binary_creator_` to create methods for `paddle.Tensor`. They are imported as-is and attached to `paddle.Tensor` as methods.

Also, many operations listed are not binary (like abs, acos, ...). I suggest changing the name.

**Review by iclementine — CHANGES_REQUESTED** [review]
Some functions that are attached to paddle.Tensor as methods are not binary operation function. 

It's better to change the name.

File: python/paddle/tensor/math.py (hunk 3 above)
Comment (zhiboniu) [reply python/paddle/tensor/math.py:402]: 因为跟本次修改内容无关，且修改目的不同。所以先记录，本pr里不针对修改。

File: python/paddle/tensor/math.py (hunk 4 above)
Comment (zhiboniu) [reply python/paddle/tensor/math.py:403]: 因为跟本次修改内容无关，且修改目的不同。所以先记录，本pr里不针对修改。

File: python/paddle/fluid/layers/math_op_patch.py (hunk 5 above)
Comment (zhiboniu) [reply python/paddle/fluid/layers/math_op_patch.py]: done, binary_ops->tensor_method_func

File: python/paddle/tensor/__init__.py (hunk 6 above)
Comment (zhiboniu) [reply python/paddle/tensor/__init__.py:185]: done, binary_ops->tensor_method_func

**Review by iclementine — DISMISSED** [review]
LGTM

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM
